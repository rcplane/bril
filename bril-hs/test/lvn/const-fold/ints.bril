@main {
  eight: int = const 8;
  two: int = const 2;

  sum: int = add eight two;
  prod: int = mul eight two;
  diff: int = sub eight two;
  quot: int = div eight two;

  print sum prod diff quot;

  eq: bool = eq eight two;
  lt: bool = lt eight two;
  gt: bool = gt eight two;
  le: bool = le eight two;
  ge: bool = ge eight two;

  print eq lt gt le ge;
}
