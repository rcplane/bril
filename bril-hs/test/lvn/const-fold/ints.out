@main {
  sum: int = const 10;
  prod: int = const 16;
  diff: int = const 6;
  quot: int = const 4;
  print sum prod diff quot;
  eq: bool = const false;
  lt: bool = const false;
  gt: bool = const true;
  le: bool = const false;
  ge: bool = const true;
  print eq lt gt le ge;
}
